### ISSUE TITLE
Not able to enable monitoring.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#21211]
Rancher version - Build from master - `af5771ff `
Steps to reproduce the problem:
Create a RKE cluster (DO cluster in my case)
Enable monitoring.
Monitoring is not getting enabled . There is no app getting deployed in system project.

Following error seen in logs:

```
2019-06-28 01:48:40.052872 I | mvcc: store.index: compact 2110
2019-06-28 01:48:40.055534 I | mvcc: finished scheduled compaction at 2110 (took 1.432934ms)
2019/06/28 01:49:09 [ERROR] ClusterController c-tplgm recovered from panic "runtime error: invalid memory address or nil pointer dereference". (err=<nil>) Call stack:
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:154
/usr/local/go/src/runtime/asm_amd64.s:522
/usr/local/go/src/runtime/panic.go:513
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/go/src/github.com/rancher/rancher/pkg/app/utils/utils.go:70
/go/src/github.com/rancher/rancher/pkg/controllers/user/monitoring/operatorHandler.go:157
/go/src/github.com/rancher/rancher/pkg/controllers/user/monitoring/operatorHandler.go:42
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/condition/condition.go:212
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/condition/condition.go:181
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/condition/condition.go:170
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/condition/condition.go:140
/go/src/github.com/rancher/rancher/pkg/controllers/user/monitoring/operatorHandler.go:40
/go/src/github.com/rancher/rancher/pkg/controllers/user/monitoring/register.go:49
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/management.cattle.io/v3/zz_generated_cluster_controller.go:144
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:348
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:258
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:246
/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:238
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1333
```

**Comment by sangeethah** [comment]
Relating to https://github.com/rancher/rancher/issues/20975

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher: master-head  b4340af9

Steps:
- run Rancher: master-head container
- add an RKE cluster ( DO, 1 etcd, 1 control plane, 3 workers)
- enable the cluster monitoring
- add a new project 
- enable the project monitoring 

Result:
- Both the cluster monitoring and the project monitoring are working as expected
